airflow/www/views.py

```diff
@@ -1499,15 +1501,23 @@ def trigger(self, session=None):
                 except TypeError:
                     flash("Could not pre-populate conf field due to non-JSON-serializable data-types")
             return self.render_template(
-                'airflow/trigger.html', dag_id=dag_id, origin=origin, conf=default_conf, doc_md=doc_md
+                'airflow/trigger.html',
+                dag_id=dag_id,
+                origin=origin,
+                conf=default_conf,
+                doc_md=doc_md,
+                form=form,
             )
 
         dag_orm = session.query(models.DagModel).filter(models.DagModel.dag_id == dag_id).first()
         if not dag_orm:
             flash(f"Cannot find dag {dag_id}")
             return redirect(origin)
 
-        execution_date = timezone.utcnow()
+        if request_execution_date is not None:
+            execution_date = timezone.parse(request_execution_date)
+        else:
+            execution_date = timezone.utcnow()
```

Comment:
i think this logic is unnecessary branching if we can default to `timezone.utcnow()` in `request.values.get` method.